File: frontend/lib/features/driver/domain/entities/driver.dart
import 'driver_availability.dart';
import 'driver_status.dart';

class Driver {
  final String id;
  final String driverId;
  final String fullName;
  final String phoneNumber;
  final String licenseNumber;
  final String ambulanceId;
  final DriverAvailability availabilityStatus;
  final DriverStatus currentStatus;
  final DateTime createdAt;
  final DateTime updatedAt;

  Driver({
    required this.id,
    required this.driverId,
    required this.fullName,
    required this.phoneNumber,
    required this.licenseNumber,
    required this.ambulanceId,
    required this.availabilityStatus,
    required this.currentStatus,
    required this.createdAt,
    required this.updatedAt,
  });
}

File: backend/app/models/driver.py
from typing import Optional
from datetime import datetime
from pydantic import BaseModel, Field

class DriverModel(BaseModel):
    id: Optional[str] = Field(None, alias="_id")
    driverId: str
    fullName: str
    phoneNumber: str
    licenseNumber: str
    ambulanceId: str
    availabilityStatus: str
    currentStatus: str
    createdAt: datetime
    updatedAt: datetime
    
    class Config:
        populate_by_name = True
